pkg/controller/jenkins/configuration/user/reconcile.go

```diff
@@ -5,45 +5,48 @@ import (
 
 	"github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2"
 	jenkinsclient "github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/client"
+	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/configuration"
 	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/configuration/backuprestore"
 	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/configuration/base/resources"
 	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/configuration/user/casc"
 	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/configuration/user/seedjobs"
 	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/groovy"
+	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/notifications"
 
 	"github.com/go-logr/logr"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	k8s "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
 // ReconcileUserConfiguration defines values required for Jenkins user configuration
 type ReconcileUserConfiguration struct {
-	k8sClient     k8s.Client
+	configuration.Configuration
 	jenkinsClient jenkinsclient.Jenkins
 	logger        logr.Logger
-	jenkins       *v1alpha2.Jenkins
-	clientSet     kubernetes.Clientset
 	config        rest.Config
 }
 
 // New create structure which takes care of user configuration
-func New(k8sClient k8s.Client, jenkinsClient jenkinsclient.Jenkins, logger logr.Logger,
-	jenkins *v1alpha2.Jenkins, clientSet kubernetes.Clientset, config rest.Config) *ReconcileUserConfiguration {
+func New(client client.Client, jenkinsClient jenkinsclient.Jenkins, logger logr.Logger,
+	jenkins *v1alpha2.Jenkins, clientSet kubernetes.Clientset, config rest.Config, notificationEvents *chan notifications.Event) *ReconcileUserConfiguration {
 	return &ReconcileUserConfiguration{
-		k8sClient:     k8sClient,
+		Configuration: configuration.Configuration{
+			Client:        client,
+			ClientSet:     clientSet,
+			Notifications: notificationEvents,
+			Jenkins:       jenkins,
+		},
```

Comment:
instead of passing client, clientset, notificationEvents and jenkins pass single Configuration parameter